Source/SwiftLintBuiltInRules/Rules/Style/IndentationStyleRule.swift

```diff
@@ -0,0 +1,385 @@
+                \tlet tabLine = true
+                \tlet spaceLine = false
+
+                \tfunc barb(arg: String) {
+                ↓ \t\tprint(arg)
+                \t}
+                }
+                """,
+                testWrappingInComment: false),
+            Example(
+                """
+                abcd(bar: "far",
+                \t baz: 0,
+                ↓     bat: false)
+                """,
+                testWrappingInComment: false),
+            Example(
+                """
+                abcd(
+                    bar: "far",
+                ↓\tbaz: 0,
+                ↓\tbat: false)
+                """,
+                testWrappingInComment: false),
+            Example(
+                """
+                // - tab_width: 4
+                abcd(bar: "far",
+                \t↓     baz: 0,
+                \t↓ bat: false)
+                """,
+                configuration: IndentationStyleConfiguration.testTabWidth,
+                testWrappingInComment: false),
+            Example(
+                """
+                // - include_multiline_strings: true
+                class Foo {
+                \tlet multiLineString = \"""
+                \t\tcontent
+                \t\t↓    more content
+                \t\t↓  most content
+                \t\t↓  \ttabbed content
+                \t\t\"""
+                }
+                """,
+                configuration: IndentationStyleConfiguration.testMultilineString,
+                testWrappingInComment: false)
+        ]
+    )
+
+    // MARK: - Initializers
+    // MARK: - Methods: Validation
+    func validate(file: SwiftLintFile) -> [StyleViolation] { // swiftlint:disable:this cyclomatic_complexity
+        var violations: [StyleViolation] = []
+
+        var fileStyle: ConfigurationType.PreferredStyle?
+        if configuration.perFile == false {
+            fileStyle = configuration.preferredStyle
+        }
+
+        for line in file.lines {
+            let indentationForThisLine = getIndentation(for: line)
+            let indentationCharacterCount = indentationForThisLine.count
+
+            guard indentationCharacterCount > 0 else { continue }
+
+            if
+                ignoreMultilineStrings(line: line, in: file) ||
+                ignoreMultilineComments(line: line, in: file)
+            { continue }
+
+            guard let firstLineIndentation = indentationForThisLine.first else { continue }
+
+            let confirmedFileStyle: ConfigurationType.PreferredStyle
+            if let fileStyle {
+                confirmedFileStyle = fileStyle
+            } else {
+                switch firstLineIndentation {
+                case " ":
+                    fileStyle = .spaces
+                    confirmedFileStyle = .spaces
+                case "\t":
+                    fileStyle = .tabs
+                    confirmedFileStyle = .tabs
+                default:
+                    queuedFatalError(
```

Comment:
Prefer to report another violation instead of crashing.

Reply:
Oooh smart! Blind spot there. Thanks!